pkg/services/sit_address_update/sit_address_update_request_approver.go

```diff
@@ -113,6 +114,25 @@ func (f *sitAddressUpdateRequestApprover) approveSITAddressUpdateRequest(appCtx
 			return err
 		}
 
+		//Update SIT Destination Fuel Charge to the correct ZIP
+		relatedDestinationSITFuelCharge, err := models.FetchRelatedDestinationSITFuelCharge(txnAppCtx.DB(), serviceItem.ID)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				txnAppCtx.Logger().Error("When attempting to update the SIT Destination Fuel Service Charge, no associated FSC was found.")
+			} else {
+				return err
+			}
+		}
+		serviceItemFSC, err := models.FetchServiceItem(txnAppCtx.DB(), relatedDestinationSITFuelCharge.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = f.updateServiceItemFinalAddress(txnAppCtx, serviceItemFSC, sitAddressUpdateRequest)
+		if err != nil {
+			return err
+		}
+
```

Comment:
I think adding a check before doing the fetching and updating for `relatedDestinationSITFuelCharge` should resolve the test cases that are failing. My suggestion should be add a variable like shown below or something similar.


```suggestion
		relatedDestinationSITFuelCharge, err := models.FetchRelatedDestinationSITFuelCharge(txnAppCtx.DB(), serviceItem.ID)
		relatedDestinationFSCAvailable := true
		if err != nil {
			if errors.Is(err, sql.ErrNoRows) {
				relatedDestinationFSCAvailable = false
				txnAppCtx.Logger().Error("When attempting to update the SIT Destination Fuel Service Charge, no associated FSC was found.")
			} else {
				return err
			}
		}
		if relatedDestinationFSCAvailable {
			serviceItemFSC, err := models.FetchServiceItem(txnAppCtx.DB(), relatedDestinationSITFuelCharge.ID)
			if err != nil {
				return err
			}

			_, err = f.updateServiceItemFinalAddress(txnAppCtx, serviceItemFSC, sitAddressUpdateRequest)
			if err != nil {
				return err
			}
		}

```

Reply:
Are you seeing a specific place where it says that's the issue? .. because the errors I'm seeing, I'm not sure this would cause that error.. also, I made a variation of the changes and still cannot get the # of server errors to reduce.